Versioning/depency to carrierwave and updating translations for carrierwave 3 (#3 and #25)

araccaine opened the pull request:
## Summary

This PR addresses #3 and #25 as they are both strongly related.

Please feel free to comment, revise and update this PR.

## What

### Tests

Two tests failed when running `$ rspec` because the capitalization was wrong:  *translation* instead of *Translation*. I updated these characters.

### Versioning

- Added `carrierwave` version 3 as runtime dependency and use `3.0.0` as new version for this gem as proposed in #3 

### Translations and keys

See #25

- Updated the key names to reflect changes in [carrierwave 3](https://github.com/carrierwaveuploader/carrierwave/blob/master/lib/carrierwave/locale/en.yml) (see #25)
    - `_whitelist_` to `_allowlist_`
    - `_blacklist_` to `_denylist_`
- Merged the `_processing_error` keys into a generic one and updated the translations if possible
    - Removed the part where the error is added to the translation (... `%{e}` ...)
    - Removed the parts with *rmagick* / *imagemagick* strings were I could safely do so. Please note: Japanese `ja` and Thai `th` were not updated because I couldn't update and verify these strings for correctness
- Added translations for the new keys if possible
   - German (`de`) by myself
    - :heart: to @Jasonyang1991 for the new Chinese translations (`zh-TW` and `zh-CN`) and for enhancing the existing ones

## Further comments

One may also create a new branch like `carrierwave-2-x` from the current `carrierwave-i18n` version `0.2.0` and update the `gemspec` file there as well for consistency? Maybe also updating the version number to `2.2.0` to be consistent?

Maybe also the pipeline for publishing on rubygems.org needs an update to allow independent managing of this gem?
E.g. the updated chinese translations may also go to version 0.2.0 e.g. as new "carrierwave 2 dependent version" 0.2.1.

mshibuya commented:
How wonderful, thank you so much for working on such a heavy lifting!
I'll prepare a gem release for CarrierWave pre-3.x versions, please wait for a moment.

mshibuya commented on rails/locales/ja.yml:
```diff
@@ -4,11 +4,10 @@ ja:
+      extension_denylist_error: "%{extension}ファイルのアップロードは許可されていません。アップロードできないファイルタイプ: %{prohibited_types}"
+      content_type_allowlist_error: "%{content_type}ファイルのアップロードは許可されていません。アップロードできるファイルタイプ: %{allowed_types}"
+      content_type_denylist_error: "%{content_type}ファイルのアップロードは許可されていません"
+      processing_error: "rmagickがファイルを処理できませんでした。画像を確認してください。" # rmagick part has to be removed
```
I speak Japanese and this will be:

```suggestion
      processing_error: "ファイルを処理できませんでした。画像を確認してください。"
```

mshibuya commented on rails/locales/th.yml:
```diff
@@ -4,11 +4,10 @@ th:
+      extension_denylist_error: "ไม่อนุญาติให้อัพโหลดไฟล์นามสกุล %{extension}, ไม่อนุญาตไฟล์ประเภท: %{prohibited_types}"
+      content_type_allowlist_error: "ไม่อนุญาติให้อัพโหลดไฟล์ประเภท %{content_type}"
+      content_type_denylist_error: "ไม่อนุญาติให้อัพโหลดไฟล์ประเภท %{content_type}"
+      processing_error: "จัดการด้วย rmagick ผิดพลาด, ไฟล์อาจจะไม่ใช่รูปภาพ?" # rmagick part has to be removed
```
I don't understand Thai but maybe it will be like this, as `ด้วย` seems to stand for 'with'?

```suggestion
      processing_error: "จัดการผิดพลาด, ไฟล์อาจจะไม่ใช่รูปภาพ?"
```

araccaine commented:
> How wonderful, thank you so much for working on such a heavy lifting! I'll prepare a gem release for CarrierWave pre-3.x versions, please wait for a moment.

Thank you, too :)

Regarding the merge conflict in `lib/carrierwave-i18n/version.rb`:  
I rebased my branch onto your latest changes and updated the Korean locale as well. Just to be clear:

- Version `0.3.0` is the updated gem version for Carrierwave 1 and 2
- Version `3.0.0` is the gem version for Carrierwave 3

Did I got this right? Otherwise feel free to correct this :)

araccaine commented:
I think you need at least Ruby 2.5 for the automated test with Ruby 2:

```
Bundler could not find compatible versions for gem "ruby": 
  In Gemfile: 
    ruby 
      carrierwave-i18n was resolved to 3.0.0, which depends on 
        carrierwave (< 4, >= 3.0.0) was resolved to 3.0.3, which depends on
          ruby (>= 2.5.0)
```

mshibuya commented:
> Version 0.3.0 is the updated gem version for Carrierwave 1 and 2
> Version 3.0.0 is the gem version for Carrierwave 3

That's correct! I think it's good to go now.
Thank you for the amazing contribution 🌟

mshibuya commented:
@araccaine  Just released 3.0.0 🚀 
By the way, are you interested in having the committer access to this repo? I think you now know this project very well...

araccaine commented:
> @araccaine Just released 3.0.0 🚀

Nice :partying_face: 

> By the way, are you interested in having the committer access to this repo? I think you now know this project very well...

Yes I would be glad to do so :)

mshibuya commented:
Great! The invitation is on the way 👍
